### PR TITLE
Oss doc

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Moving #63 to the right direction, hopefully.

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by blefaudeux** [comment]
note: I resurrected an old branch for that, and did not realize that the old commits would show up here

**Review by msbaines — COMMENTED** [review]
Docs stuff looks OK. Can you rebase or merge with master to avoid extraneous diffs due to changes to benchmarks/oss.py

**Review by Vittorio-Caggiano — APPROVED** [review]

**Comment by blefaudeux** [comment]
@msbaines the benchmark changes were on purpose actually, fixing the broken all_reduce/loss.backward() + minor readability. I hope that works for you, landed to get some of the API documentation up ?
